.ci.yaml

```diff
@@ -10,6 +10,23 @@ enabled_branches:
   - flutter-\d+\.\d+-candidate\.\d+
 
 platform_properties:
+  linux_staging_build:
```

Comment:
I am a bit reluctant to add this set of builders in the .ci.yaml.
Instead, after taking a look at the failure: https://ci.chromium.org/ui/p/flutter/builders/staging/Linux_staging%20analyze/62/overview
you are missing the `validation` property: https://github.com/flutter/flutter/blob/master/.ci.yaml#L222-L223

Reply:
Shouldn't we be moving to this scheme and deprecating running staging tests with devicelab_staging_config.star, to avoid having to maintain 2 different methods of running staging tests, and avoid issues like this where config is different?

If we choose to keep devicelab_staging_config.star as a way of testing staging tests, we will continue to experience inconsistencies and overhead with our main ci.yaml based mechanism of defining builds.

Reply:
+1 to moving it into the ci.yaml. I think some ways to simplify it is we could update the ci.yaml infra to instead not require an explicit staging platform_properties, but other than that this approach LGTM

Reply:
How do we validate all Linux builders (debian->ubuntu) work without affecting our end users? With this, we need to add ~100 new entries to the dashboard and then remove all of them.

Reply:
Your correct, right now, the view is going to be messy when we do this. And we should probably find a more graceful way of handling the build view affects in the future (maybe hide flaky by default, or even have a `staging:true` property that hides only the staging ones). 

However, I think utilizing this same mechanism thats used for prod builds to schedule these staging ones are good to do longterm.

Reply:
That sounds fair. The ideal case if we can easily control what targets to be enabled in prod (show up in dashboard), what targets to to enabled in `bringup: true` (show up in dashboard), and what targets to be enabled in staging for validation purpose only (hide from dashboard), using a single source of truth of the .ci.yaml.

But these need backend refactoring to support. Before we have a clear plan/support, I recommend use existing staging star to validate.

Reply:
I still think the easiest way would be to add `os: Ubuntu` to targets we want to individually migrate, and do it in batches. Once everything is swapped, we can update the platform_properties to default to ubuntu.

Reply:
Switching it that way doesn't allow provide the ability to let them soak for a while. Having the ability to setup staging jobs and let them run as we need would be very helpful. Even if we have to do them in blocks.